BanchoSharp/Multiplayer/MultiplayerLobby.cs

```diff
@@ -417,12 +417,19 @@ private void UpdatePlayerInformation(string banchoResponse)
 		// Bancho may send extra player info after the name, for example "[Host / HardRock]", after the 16
 		// character player name bit.
 		string? playerInfo = banchoResponse.Length > (playerNameBegin + 16) ? banchoResponse[(playerNameBegin + 16)..] : null;
-
+		
+		// Find the digits from "/u/" to where the name begins
+		int playerId = int.Parse(banchoResponse[banchoResponse.IndexOf("/u/", StringComparison.Ordinal)..(playerNameBegin-1)].Where(char.IsDigit).ToArray());
+		
 		var player = FindPlayer(playerName);
 
 		if (player is null)
 		{
-			player = new MultiplayerPlayer(playerName, slot);
+			player = new MultiplayerPlayer(playerName, slot)
+			{
+				Id = playerId
```

Comment:
We can save a few lines: no need to specify the Id on init if we are force setting it anyway.

Reply:
Then when `OnPlayerJoined` gets invoked, the player doesn't have a valid id, even though we already have the data.